### PR TITLE
ydiff: init at 1.1

### PR DESCRIPTION
**Pull request by leenaars**
###### Motivation for this change

Handy tool for working with diffs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by teto** [comment]
There is no homepage (in meta) ?

**Comment by leenaars** [comment]
Added the GitHub page. With fetchFromGitHub that would have been automatic, but of course that doesn't hold.

**Comment by Infinisil** [comment]
@GrahamcOfBorg build ydiff

**Comment by Infinisil** [comment]
> Ran 0 tests in 0.000s

Are you sure the tests get run?

**Comment by leenaars** [comment]
The tests weren't being run, and for completeness sake I've added them.

File: pkgs/development/tools/ydiff/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      --replace "pep8 --ignore" "#pep8 --ignore" \
+      --replace "python3 \`which coverage\`" "${python3Packages.coverage}/bin/coverage3"
+  '';
+
+  buildInputs = [ docutils pygments ];
+  propagatedBuildInputs = [ less patchutils ];
+  checkInputs = [ coverage coreutils git subversion which ];
+
+  checkPhase = lib.optional (!isPy3k) "make test" ++
+               lib.optional (isPy3k) "make test3";
```
Comment (Infinisil) [inline pkgs/development/tools/ydiff/default.nix]: You can just use
```nix
checkPhase = if isPy3k then ''
  make test3
'' else ''
  make test
''
```

**Review by infinisil — CHANGES_REQUESTED** [review]

**Comment by Infinisil** [comment]
@GrahamcOfBorg build ydiff

File: pkgs/development/tools/ydiff/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      --replace "pep8 --ignore" "#pep8 --ignore" \
+      --replace "python3 \`which coverage\`" "${python3Packages.coverage}/bin/coverage3" \
+      --replace /bin/sh ${stdenv.shell}
+  '';
+
+  buildInputs = [ docutils pygments ];
+  propagatedBuildInputs = [ less patchutils ];
+  checkInputs = [ coverage coreutils git subversion which ];
+
+  checkPhase = if isPy3k then ''make test3'' else ''make test'';
```
Comment (Infinisil) [inline pkgs/development/tools/ydiff/default.nix]: Sorry to nitpick, but if it's in a single line, then the `''` quotes aren't needed:
```nix
{
  checkPhase = if isPy3k then "make test3" else "make test";
}
```

:D

Comment (leenaars) [reply pkgs/development/tools/ydiff/default.nix]: Indeed, so let's shave off those two extra characters...

Comment (leenaars) [reply pkgs/development/tools/ydiff/default.nix]: There is something rewarding with a tool that you can use on the commit itself. 

ydiff -l -s

**Comment by Infinisil** [comment]
@GrahamcOfBorg build ydiff

**Comment by Infinisil** [comment]
Can't you test this locally?

**Comment by leenaars** [comment]
Annoyingly, it builds on my system - I only push when it does that. But then there is different behaviour on the CI I'm trying to fix. Apologies!

**Comment by Infinisil** [comment]
Try enabling the sandbox for your builds

File: pkgs/development/tools/ydiff/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      --replace /bin/sh ${stdenv.shell} \
+      --replace tests/regression.sh "${stdenv.shell} tests/regression.sh"
+    patchShebangs tests/*.sh
+  '';
+
+  buildInputs = [ docutils pygments ];
+  propagatedBuildInputs = [ less patchutils ];
+  checkInputs = [ coverage coreutils git subversion which ];
+
+  checkPhase = if isPy3k then "make test3" else "make test";
```
Comment (matthewbauer) [inline pkgs/development/tools/ydiff/default.nix]: You can use checkTarget instead of checkPhase here.

Comment (leenaars) [reply pkgs/development/tools/ydiff/default.nix]: @matthewbauer: thanks, didn't know about that. Did that.

File: pkgs/development/tools/ydiff/default.nix (hunk 1 above)
Comment (leenaars) [reply pkgs/development/tools/ydiff/default.nix]: Now using checkTarget, as suggested.

**Review by infinisil — APPROVED** [review]

**Pull request merged**